_posts/2023-01-22-auto-restarting-connectors.md

```diff
@@ -0,0 +1,317 @@
+---
+layout: post
+title:  "Auto-restarting connectors in Apache Kafka Connect and Mirror Maker 2"
+date: 2023-01-22
+author: jakub_scholz
+---
+
+Apache Kafka Connect provides a framework for integrating Apache Kafka with external systems using connectors.
+Connectors run inside the Kafka Connect deployment, connect to the external system, and push data from Kafka to the external system or the other way around.
+When running Kafka Connect, you have to monitor not only the state of the Kafka Connect deployment, but also the state of the connectors and their tasks.
+If a connector or one of its tasks fail, you check the reason it failed so you know how to fix it.
+In many cases, the issue is something temporary like a network glitch or an outage of the external system.
+And all you need to do to fix it is just restart the connector or task.
+Could this be something the operator might do for you?
+
+<!--more-->
+
+Having the operator automatically restart failed connectors or their tasks was one of the most common feature requests received by Strimzi.
+And in Strimzi 0.33, this feature was contributed by [Thomas Dangleterre](https://github.com/ThomasDangleterre).
```

Comment:
Is Thomas OK with his name appearing in this blog post?

Reply:
Yes.